### PR TITLE
[Slider] Fix slider label not moving

### PR DESCRIPTION
**Pull request by manabu-nakamura**
#3124
https://github.com/material-components/material-components-android/issues/2869#issuecomment-1821123500

### CODE REVIEW

File: lib/java/com/google/android/material/slider/BaseSlider.java [hunk 1]
```diff
@@ -361,6 +361,8 @@ abstract class BaseSlider<
           }
         }
       };
+  @NonNull
+  private final ViewTreeObserver.OnGlobalLayoutListener onGlobalLayoutListener = onScrollChangedListener::onScrollChanged;
```
Comment (drchen) [inline lib/java/com/google/android/material/slider/BaseSlider.java:365]: This is sort of confusing. Can we declare a method, say, `updateLableVisibility()`, and do something like:

```
private final ViewTreeObserver.OnScrollChangedListener onScrollChangedListener = this::updateLableVisibility;
private final ViewTreeObserver.OnGlobalLayoutListener onGlobalLayoutListener = this::updateLableVisibility;
```

instead?

Comment (manabu-nakamura) [reply lib/java/com/google/android/material/slider/BaseSlider.java]: Thank you very much. I tried. I'm not sure if the name of the method is appropriate.

File: lib/java/com/google/android/material/slider/BaseSlider.java [hunk 2]
```diff
@@ -341,26 +341,9 @@ abstract class BaseSlider<
...
-              ViewUtils.getContentViewOverlay(this).add(label);
-            } else {
-              ViewUtils.getContentViewOverlay(this).remove(label);
-            }
-          }
-        }
-      };
+  private final ViewTreeObserver.OnScrollChangedListener onScrollChangedListener = this::updateLabel;
+  @NonNull
+  private final ViewTreeObserver.OnGlobalLayoutListener onGlobalLayoutListener = this::updateLabel;
```
Comment (dsn5ft) [inline lib/java/com/google/android/material/slider/BaseSlider.java:346]: do we need both the global layout listener and the scroll listener or is just the global layout listener enough?

Comment (manabu-nakamura) [reply lib/java/com/google/android/material/slider/BaseSlider.java:346]: By only the global layout listener, the slider label doesn't move while scrolling (#2869).

**Pull request closed without merging**